src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -324,45 +325,51 @@ def _validate_service_principal(client, sp_id):
 
 
 def _build_service_principal(client, name, url, client_secret):
-    sys.stdout.write('creating service principal')
+    view = progress.DeterminateStandardOut()
```

Comment:
Because it is not supposed to be a component exposed to a general use. We may introduce other types of views in the future, retired views or change views logic.